lib/cc/analyzer/formatters/plain_text_formatter.rb

```diff
@@ -56,7 +56,11 @@ def finished
 
         def engine_running(engine)
           @active_engine = engine
-          with_spinner("Running #{engine.name}: ") do
+          if $stdout.tty?
+            with_spinner("Running #{engine.name}: ") do
+              yield
+            end
+          else
             yield
```

Comment:
I think I suggested this before, but you can change the syntax here:

```rb
def engine_running(engine, &block)
  # ...
  with_spinner("...", &block)
end
```

In this case I think it'd help because you're pushing another level of indentation.